npm/src/upload/index.ts

```diff
@@ -51,12 +55,15 @@ export class UploadFiles {
             throw Error("No files selected")
           }
 
-          this.clientFileProcessing.processClientFiles(
+          await this.clientFileProcessing.processClientFiles(
             consignmentId,
             files,
             progress => console.log(progress.percentageProcessed),
             this.stage
           )
+          if (uploadDataFormRedirect) {
```

Comment:
Does the redirect happen, only if the await function completes?

Reply:
Yeah, it should only redirect if the upload has successfully finished not if it errors out or stalls for any reason :raised_hands: